packages/fuels/tests/providers.rs

```diff
@@ -308,46 +286,19 @@ async fn contract_deployment_respects_maturity() -> Result<()> {
             LoadConfiguration::default(),
         )
         .map(|loaded_contract| {
-            loaded_contract.deploy(wallet, TxParameters::default().with_maturity(maturity))
+            loaded_contract.deploy(wallet, TxPolicies::default().with_maturity(maturity))
         })
     };
 
-    let err = deploy_w_maturity(1u32)?.await.expect_err("Should not have been able to deploy the contract since the block height (0) is less than the requested maturity (1)");
-    assert!(matches!(
-        err,
-        Error::ValidationError(fuel_tx::CheckError::TransactionMaturity)
-    ));
+    let err = deploy_w_maturity(1u32)?.await.expect_err(
+        "Should not deploy contract since block height (0) is less than the requested maturity (1)",
+    );
+    assert!(matches!(err, Error::ValidationError(s) if s == "TransactionMaturity"));
```

Comment:
Although it is easier to write, we should probably try and avoid `assert!(matches!` assertions -- they don't make for great diagnostics when the test fails -- you just get that an assertion had failed. If we had done something like this:
```rust
    let Error::ValidationError(s) = err else {
        panic!("Expected a ValidationError, got: {err}");
    };
    assert_eq!(s, "TransactionMaturity");
```

The error would be much clearer. Maybe there is a crate somewhere for better messages when matching against enums. In any case I don't suggest you change anything, just thought I would share.